tests/irdl/test_declarative_assembly_format.py

```diff
@@ -3155,3 +3156,155 @@ def test_multiple_operand_extraction_fails():
         "Possible values are: {i32, index}",
     ):
         parser.parse_operation()
+
+
+################################################################################
+#                                  IntAttr                                     #
+################################################################################
+
+
+@pytest.mark.parametrize(
+    "program",
+    ["%0 = test.int_attr_extract 1", "%0, %1 = test.int_attr_extract 2"],
+)
+def test_int_attr_extraction(program: str):
+    @irdl_op_definition
+    class IntAttrExtractOp(IRDLOperation):
+        name = "test.int_attr_extract"
+
+        _I: ClassVar = IntVarConstraint("I", AnyInt())
+
+        prop = prop_def(
+            IntegerAttr.constr(value=IntAttrConstraint(_I), type=eq(IndexType()))
+        )
+
+        outs = var_result_def(RangeOf(eq(IndexType()), length=_I))
+
+        assembly_format = "$prop attr-dict"
+
+    ctx = MLContext()
+    ctx.load_op(IntAttrExtractOp)
+
+    check_roundtrip(program, ctx)
+
+
+@pytest.mark.parametrize(
+    "program, error",
+    [
+        (
+            "%0 = test.int_attr_extract 2",
+            "Operation has 2 results, but was given 1 to bind",
+        ),
+        (
+            "%0, %1 = test.int_attr_extract 1",
+            "Operation has 1 results, but was given 2 to bind",
+        ),
+    ],
+)
+def test_int_attr_extraction_errors(program: str, error: str):
+    @irdl_op_definition
+    class IntAttrExtractOp(IRDLOperation):
+        name = "test.int_attr_extract"
+
+        _I: ClassVar = IntVarConstraint("I", AnyInt())
+
+        prop = prop_def(
+            IntegerAttr.constr(value=IntAttrConstraint(_I), type=eq(IndexType()))
+        )
+
+        outs = var_result_def(RangeOf(eq(IndexType()), length=_I))
+
+        assembly_format = "$prop attr-dict"
+
+    ctx = MLContext()
+    ctx.load_op(IntAttrExtractOp)
+    parser = Parser(ctx, program)
+    with pytest.raises(ParseError, match=error):
+        parser.parse_optional_operation()
+
+
+@pytest.mark.parametrize(
+    "program",
+    [
+        "test.int_attr_verify 1, %0",
+        "test.int_attr_verify 2, %0, %1",
+        "test.int_attr_verify 1 and 1, %0",
+        "test.int_attr_verify 2 and 2, %0, %1",
+    ],
+)
+def test_int_attr_verify(program: str):
+    @irdl_op_definition
+    class IntAttrExtractOp(IRDLOperation):
+        name = "test.int_attr_verify"
```

Comment:
Same here, plus a name for the class and its name that match (here one uses 'extract' and the other uses 'verify'